processing/src/main/java/org/apache/druid/segment/ObjectColumnSelector.java

```diff
@@ -19,75 +19,50 @@
 
 package org.apache.druid.segment;
 
+import org.apache.druid.error.DruidException;
+
 /**
- * This class is convenient for implementation of "object-sourcing" {@link ColumnValueSelector}s, it provides default
- * implementations for all {@link ColumnValueSelector}'s methods except {@link #getObject()} and {@link
- * #classOfObject()}.
- *
- * This class should appear ONLY in "extends" clause or anonymous class creation, but NOT in "user" code, where
- * {@link BaseObjectColumnValueSelector} must be used instead.
+ * Object based column selector.
  *
- * This is a class rather than an interface unlike {@link LongColumnSelector}, {@link FloatColumnSelector} and {@link
- * DoubleColumnSelector} solely in order to make {@link #isNull()} method final.
+ * This abstract class provides a {@link ColumnValueSelector} based on the methods of {@link BaseObjectColumnValueSelector}.
  */
 public abstract class ObjectColumnSelector<T> implements ColumnValueSelector<T>
 {
-  /**
-   * @deprecated This method is marked as deprecated in ObjectColumnSelector to minimize the probability of accidental
-   * calling. "Polymorphism" of ObjectColumnSelector should be used only when operating on {@link ColumnValueSelector}
-   * objects.
-   */
-  @Deprecated
+  private static final String COLUMN_IS_NULL_ERROR_MESSAGE = "Invalid usage pattern: method returning primitive called - but the column is null";
+
   @Override
-  public float getFloat()
+  public final float getFloat()
   {
     T value = getObject();
     if (value == null) {
-      return 0;
+      throw DruidException.defensive(COLUMN_IS_NULL_ERROR_MESSAGE);
     }
     return ((Number) value).floatValue();
   }
 
-  /**
-   * @deprecated This method is marked as deprecated in ObjectColumnSelector to minimize the probability of accidental
-   * calling. "Polymorphism" of ObjectColumnSelector should be used only when operating on {@link ColumnValueSelector}
-   * objects.
-   */
-  @Deprecated
   @Override
-  public double getDouble()
+  public final double getDouble()
   {
     T value = getObject();
     if (value == null) {
-      return 0;
+      throw DruidException.defensive(COLUMN_IS_NULL_ERROR_MESSAGE);
```

Comment:
Is it possible to add an unit test as mentioned in the issue

Reply:
I've extended `windowed_long_null.sqlTest` a bit to cover  one there too - but the drill related cases were also been fixed by this change